compiler/luci/pass/src/InsertQuantizeOpOnDTypeMismatch.cpp

```diff
@@ -119,6 +119,51 @@ luci::CircleQuantize *create_quantize_op(luci::CircleNode *node)
 namespace luci
 {
 
+// Add Quantize Op before Transpose
+//
+// Before
+//
+//     [Node] s16
+//        |
+//  [Transpose] u8
+//
+// After
+//
+//     [Node] s16
+//        |
+//   [Quantize] s16->u8
+//        |
+//  [Transpose] u8
+void InsertQuantizeOpOnDTypeMismatch::visit(luci::CircleTranspose *node)
+{
+  auto input = loco::must_cast<luci::CircleNode *>(node->a());
+
+  // Input dtype == Output dtype. No problem
+  if (input->dtype() == node->dtype())
+    return;
+
+  // Only cares quantized case
+  if (not is_quantized(input))
+    return;
+
+  if (not is_quantized(node))
+    return;
```

Comment:
I think these lines follows other implementations. But, just curiosity, can this case really happen?

Reply:
I expect that it should not happen for our target model, but I'm not 100% sure. These branches are useful because I don't have to consider whether this case will never happen or not. We can just apply this pass to safe cases.